### PR TITLE
Fix revealing fakes by double-clicking a building

### LINKED ISSUE #8181: France, reveal fake buildings easily!
If you double click (to highlight all of same type) on a fake building, it won't also highlight the real one revealing it to enemies easily. Works the other way around too, double click real building doesn't highlight fake ones.

Found by me, confirmed by Penev.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/15448174-france-reveal-fake-buildings-easily?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by obrakmann**
Fixes #8181

### CODE REVIEW

File: OpenRA.Game/Widgets/WorldInteractionControllerWidget.cs [hunk 1]
```diff
@@ -110,7 +110,8 @@ public override bool HandleMouseInput(MouseInput mi)
 							.WithHighestSelectionPriority();
 
 						var newSelection2 = SelectActorsInBox(World, worldRenderer.Viewport.TopLeft, worldRenderer.Viewport.BottomRight,
-							a => unit != null && a.Info.Name == unit.Info.Name && a.Owner == unit.Owner);
+							a => unit != null && World.RenderPlayer != null && unit.Owner == World.RenderPlayer
```
Comment (penev92) [inline OpenRA.Game/Widgets/WorldInteractionControllerWidget.cs]: `&& World.RenderPlayer != null` is not necessary.
If it is `null`, then `unit.Owner == World.RenderPlayer` would just be false.

Comment (deniz1a) [inline OpenRA.Game/Widgets/WorldInteractionControllerWidget.cs]: Can unit.Owner be null?

**Comment by epicelite** [comment]
Huzzah! Thanks obrakmann!

**Comment by Mailaender** [comment]
This will turn off double click actor selection with the shroud disable cheat activated.

**Comment by pchote** [comment]
Could we fix that with a `RenderPlayer ?? LocalPlayer`, like we did for the team health bars?

**Comment by obrakmann** [comment]
Updated.

**Comment by Mailaender** [comment]
Works as promised. :white_check_mark:

**Comment by penev92** [comment]
Seems legit :+1:

**Pull request merged**